### PR TITLE
Defaut ACL:Read only

### PR DESCRIPTION
**Pull request by robyj**
- Modified root-authentication.ttl
- Modified WebAC integration tests
- Modified code around getAgentRoles() and getTypes()

Resolves https://jira.duraspace.org/browse/FCREPO-2668

**The title of this pull-request should be a brief description of what the pull-request fixes/improves/changes. Ideally 50 characters or less.**
* * *

**JIRA Ticket**: (link)

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)

# What does this Pull Request do?
A brief description of what the intended result of the PR will be and/or what problem it solves.

# What's new?
A in-depth description of the changes made by this PR. Technical details and possible side effects.

Example:
* Changes x feature to such that y
* Added x
* Removed y

# How should this be tested?

A description of what steps someone could take to:
* Reproduce the problem you are fixing (if applicable)
* Test that the Pull Request does what is intended.
* Please be as detailed as possible.
* Good testing instructions help get your PR completed faster.

# Additional Notes:
Any additional information that you think would be helpful when reviewing this PR.

Example:
* Does this change require documentation to be updated? 
* Does this change add any new dependencies? 
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? 
* Could this change impact execution of existing code?

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo4/committers

### CODE REVIEW

File: fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACAuthorizingRealm.java [hunk 1]
```diff
@@ -170,7 +170,9 @@ protected AuthorizationInfo doGetAuthorizationInfo(final PrincipalCollection pri
 
     private Map<String, Collection<String>> getRolesForPath() {
         Map<String, Collection<String>> roles = null;
-        final FedoraResource fedoraResource = getResourceOrParentFromPath(request.getPathInfo());
+        //robyj
```
Comment (awoods) [inline fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACAuthorizingRealm.java]: Please revert the changes to this class.

File: fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACFilter.java [hunk 2]
```diff
@@ -127,9 +127,12 @@ public void doFilter(final ServletRequest request, final ServletResponse respons
                 log.debug("User has fedoraUser role");
                 // non-admins are subject to permission checks
                 if (!isAuthorized(currentUser, httpRequest)) {
+                    log.debug("is not authorized");
```
Comment (awoods) [inline fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACFilter.java]: Please revert the changes to this class.

File: fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java [hunk 3]
```diff
@@ -40,6 +40,7 @@
 import static org.fcrepo.auth.webac.URIConstants.WEBAC_MODE_VALUE;
 import static org.fcrepo.auth.webac.URIConstants.WEBAC_NAMESPACE_VALUE;
 import static org.fcrepo.http.api.FedoraAcl.getDefaultAcl;
+import static org.fcrepo.kernel.api.FedoraTypes.FEDORA_REPOSITORY_ROOT;
```
Comment (awoods) [inline fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java]: Please revert the changes to this class.

File: fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java [hunk 4]
```diff
@@ -825,27 +825,29 @@ public void testAgentAsUri() throws IOException {
...
+        assertEquals(HttpStatus.SC_OK, getStatus(requestGet2));
 
         System.setProperty(USER_AGENT_BASE_URI_PROPERTY, "info:user/");
         System.setProperty(GROUP_AGENT_BASE_URI_PROPERTY, "info:group/");
 
         logger.debug("Can username 'smith123' read {} (overridden system ACL)", id);
         final HttpGet requestGet3 = getObjMethod(id);
         setAuth(requestGet3, "smith123");
-        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(requestGet3));
+        // robyj - fcrepo-2668 changed this from SC_FORBIDDEN to SC_OK
```
Comment (awoods) [inline fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java]: Comment can be removed.

File: fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java [hunk 5]
```diff
@@ -825,27 +825,29 @@ public void testAgentAsUri() throws IOException {
...
+        assertEquals(HttpStatus.SC_OK, getStatus(requestGet3));
 
         logger.debug("Can username 'group123' read {} (overridden system ACL)", id);
         final HttpGet requestGet4 = getObjMethod(id);
         setAuth(requestGet4, "group123");
-        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(requestGet4));
+        // robyj - fcrepo-2668 changed the code from SC_FORBIDDEN to SC_OK
```
Comment (awoods) [inline fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java]: Comment can be removed.

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraAcl.java [hunk 6]
```diff
@@ -82,6 +82,12 @@
 import org.slf4j.Logger;
 import org.springframework.context.annotation.Scope;
 
+import org.apache.jena.rdf.model.Resource;
```
Comment (awoods) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraAcl.java]: Please revert changes to this class.

File: fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java (hunk 5 above)
Comment (robyj) [reply fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java]: I believe i've made those modifications in the commit I just made

File: fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java [hunk 7]
```diff
@@ -212,6 +212,7 @@ private FedoraResource getBaseVersion(final FedoraResource resource) {
                                   effectiveRoles.computeIfAbsent(agent, key -> new HashSet<>())
                                                 .addAll(auth.getModes().stream().map(URI::toString).collect(toSet()));
                               });
+
```
Comment (awoods) [inline fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java]: Please revert changes to this file.

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraAcl.java [hunk 8]
```diff
@@ -346,7 +346,6 @@ public static Model getDefaultAcl(final String baseUri) {
 
         try (final InputStream is = FedoraAcl.class.getResourceAsStream(ROOT_AUTHORIZATION_LOCATION)) {
             LOGGER.debug("Getting root ACL from classpath: {}", ROOT_AUTHORIZATION_LOCATION);
-
```
Comment (awoods) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraAcl.java]: Please revert changes to this file.
